### PR TITLE
Welcome page

### LINKED ISSUE #22: Welcome Page
The welcome page consists of a static map background and a modal with a few sections. The sections are as follows

- Welcome text
- How It Works
- File Upload

### PR DESCRIPTION
**Pull request by mstone121**
## Overview

This PR contains a mostly functional implementation of the Welcome page including the three modal sections.

Closes #22 

### Demo

https://user-images.githubusercontent.com/8356789/184244876-6a6b377a-22fa-46af-8a88-8f80c201aa4f.mov

### Notes

Getting the drag and drop hover effect to work correctly was tricky. `dragleave` events are triggered when entering child components so the hover effect would disappear when the mouse entered one these elements. Setting `pointer-events: none` was not option because the upload button still needed to capture the `click` event. At bit of research revealed [relatedTarget](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/relatedTarget), a property of `MouseEvent`s that contains the element that the cursor has entered. Simply checking that `relatedTarget` is contained by the drop area means the `dragleave` event can be ignored.

## Testing Instructions

- Go to http://localhost:4545/ or http://localhost:4545/welcome
- Welcome modal should be visible. Learn more should be clickable and display more info when clicked
- "How It Works" page is next.
- Then File Upload page. Multiple files can be added at once, either by drag-n-dropping or by using the file picker
- After files are uploaded, they should show up on the right side with indeterminate progress bars
- Also, check that previous pages can be reached by clicking "Previous" button

## Checklist

- [x] `fixup!` commits have been squashed
- [x] `CHANGELOG.md` updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [x] `README.md` updated if necessary to reflect the changes
- [x] CI passes after rebase

### CODE REVIEW

File: src/app/src/components/WelcomeModal.js [hunk 1]
```diff
@@ -1,31 +1,61 @@
...
+    const ContinueButton = (
+        <Button
+            bg='gray.900'
+            color='white'
+            boxShadow='base'
+            onClick={goToNextSection}
+        >
+            Continue
+        </Button>
+    );
```
Comment (rajadain) [inline src/app/src/components/WelcomeModal.js:44]: I would recommend we rely on variants:

```suggestion
    const PreviousButton = (
        <Button variant='secondary' onClick={goToPreviousSection}>
            Previous
        </Button>
    );

    const ContinueButton = (
        <Button variant='cta' onClick={goToNextSection}>
            Continue
        </Button>
    );
```

And define them as done in Figma here: https://www.figma.com/file/tJPZ0lkT5C3SAyS2g2R1aA/IoW---Boundary-Sync-Tool?node-id=234%3A12160

```js
const Button = {
    baseStyle: {
        fontFamily: `'Inter', sans-serif`,
        borderRadius: '4px',
        fontWeight: 600,
    },
    variants: {
        cta: {
            bg: 'gray.900',
            color: 'white',
            _hover: {
                bg: 'gray.600',
                textDecoration: 'underline',
            },
            _disabled: {
                bg: 'gray.400',
            },
        },
        secondary: {
            bg: 'white',
            borderWidth: '1px',
            borderColor: 'gray.200',
            color: 'gray.600',
            _hover: {
                bg: 'gray.200',
                textDecoration: 'underline',
            },
            _disabled: {
                color: 'gray.700',
            },
        },
    },
};
```

**Review by rajadain — COMMENTED** [review]
This looks great. I'm leaving some code comments as I go through this. Going to spend more time on this review since it is an early PR and we're defining project paradigms.

I'll leave more comments shortly.

File: src/app/src/components/WelcomeModal.js [hunk 2]
```diff
@@ -1,31 +1,61 @@
...
+
+import HowItWorks from './ModalSections/HowItWorks';
+import Introduction from './ModalSections/Introduction';
+import FileUpload from './ModalSections/FileUpload';
+
+const sectionMap = {
+    0: Introduction,
+    1: HowItWorks,
+    2: FileUpload,
+};
```
[the remark this answers is not in the document]
Comment (mstone121) [reply src/app/src/components/WelcomeModal.js]: It does not. I think I was originally planning to use an `enum` before I forgot they are a Typescript feature. I squashed the fix for this into c6422d31db258a418102d1ce7c4bdf47a2df2a6e.

File: src/app/src/components/WelcomeModal.js (hunk 1 above)
Comment (mstone121) [reply src/app/src/components/WelcomeModal.js]: Good idea! 57e6ea542408a3c9dab0d1ceaf9547987fdef2e8

File: src/app/src/components/ModalSections/HowItWorks.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    );
+}
+
+function HowItWorksListItem({ heading, body }) {
+    return (
+        <>
+            <Heading size='sm' pb={1} pt={4}>
+                {heading}
+            </Heading>
+            <Text>{body}</Text>
```
Comment (rajadain) [inline src/app/src/components/ModalSections/HowItWorks.js:38]: Let's make the text a little bigger to match the prototype

```suggestion
            <Heading size='md' pb={1} pt={4}>
                {heading}
            </Heading>
            <Text fontSize='lg'>{body}</Text>
```

File: src/app/src/components/ModalSections/Introduction.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    return (
+        <ModalSection
+            heading='Welcome to Boundary Sync'
+            nextButton={
+                <Button variant='cta' onClick={goToNextSection}>
+                    Get Started
+                </Button>
+            }
+        >
+            <Text>
```
Comment (rajadain) [inline src/app/src/components/ModalSections/Introduction.js:38]: Let's add a [`textStyle`](https://chakra-ui.com/docs/styled-system/text-and-layer-styles#text-styles) here for this block:

```diff
diff --git a/src/app/src/components/ModalSections/Introduction.js b/src/app/src/components/ModalSections/Introduction.js
index bb6d28b..9cb87a7 100644
--- a/src/app/src/components/ModalSections/Introduction.js
+++ b/src/app/src/components/ModalSections/Introduction.js
@@ -35,7 +35,7 @@ export default function Introduction({ goToNextSection }) {
                 </Button>
             }
         >
-            <Text>
+            <Text textStyle='welcomeIntro'>
                 Build a digital map of your service area boundary in minutes for
                 free. No complex software or training needed.
             </Text>
diff --git a/src/app/src/theme.js b/src/app/src/theme.js
index 5a5af18..f555656 100644
--- a/src/app/src/theme.js
+++ b/src/app/src/theme.js
@@ -45,7 +45,14 @@ const theme = extendTheme({
     },
     fonts: {
         heading: `'Quicksand', sans-serif`,
-        body: `'Quicksand', sans-serif`,
+        body: `'Inter', sans-serif`,
+    },
+    textStyles: {
+        welcomeIntro: {
+            fontFamily: 'heading',
+            fontSize: 'xl',
+        },
+    },
     },
 });
 
```

File: src/app/src/components/ModalSections/ModalSection.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    heading,
+    nextButton,
+    prevButton,
+    children,
+}) {
+    return (
+        <ModalContent minH='65vh' p={8}>
+            <ModalHeader>
+                {preHeading && (
+                    <Heading size='md' color='gray.500'>
```
Comment (rajadain) [inline src/app/src/components/ModalSections/ModalSection.js:22]: Let's use a `Heading` variant to style this block:

```diff
diff --git a/src/app/src/components/ModalSections/ModalSection.js b/src/app/src/components/ModalSections/ModalSection.js
index 6265cc4..60a6d57 100644
--- a/src/app/src/components/ModalSections/ModalSection.js
+++ b/src/app/src/components/ModalSections/ModalSection.js
@@ -19,7 +19,7 @@ export default function ModalSection({
         <ModalContent minH='65vh' p={8}>
             <ModalHeader>
                 {preHeading && (
-                    <Heading size='md' color='gray.500'>
+                    <Heading as='h5' variant='preHeading'>
                         {preHeading}
                     </Heading>
                 )}
diff --git a/src/app/src/theme.js b/src/app/src/theme.js
index 5a5af18..1653108 100644
--- a/src/app/src/theme.js
+++ b/src/app/src/theme.js
@@ -39,9 +39,20 @@ const Button = {
     },
 };
 
+const Heading = {
+    variants: {
+        preHeading: {
+            color: 'gray.500',
+            fontFamily: 'body',
+            fontSize: 'md',
+        },
+    },
+};
+
 const theme = extendTheme({
     components: {
         Button,
+        Heading,
     },
     fonts: {
         heading: `'Quicksand', sans-serif`,
```

This separate the styling from the content.

**Review by rajadain — COMMENTED** [review]
Let's add this adjustment for headings as well, noticed it in the wireframe:

```diff
diff --git a/src/app/src/theme.js b/src/app/src/theme.js
index 5a5af18..4b828ea 100644
--- a/src/app/src/theme.js
+++ b/src/app/src/theme.js
@@ -47,6 +47,13 @@ const theme = extendTheme({
         heading: `'Quicksand', sans-serif`,
         body: `'Quicksand', sans-serif`,
     },
+    styles: {
+        global: {
+            'h1, h2, h3, h4, h5, h6': {
+                letterSpacing: '-0.02em',
+            },
+        },
+    },
 });
 
 export default theme;
```

File: src/app/src/components/ModalSections/Introduction.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                <LearnMoreSection />
+            </Collapse>
+        </ModalSection>
+    );
+}
+
+function LearnMoreSection() {
+    return (
+        <Box pt={4}>
+            <Heading size='sm' pb={2}>
```
Comment (rajadain) [inline src/app/src/components/ModalSections/Introduction.js:62]:
```suggestion
            <Heading size='md' pb={2}>
```

File: src/app/src/components/ModalSections/FileUpload.js [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+        <Box
+            id='drop_zone'
+            w='50%'
+            onDrop={handleUpload}
+            onDragEnter={startDrag}
+            onDragLeave={onLeaveDragBox}
+            onDragEnd={endDrag}
+            onDragOver={event => event.preventDefault()}
+            border='1px dashed'
+            borderColor={hovering ? 'gray.900' : 'gray.200'}
```
Comment (rajadain) [inline src/app/src/components/ModalSections/FileUpload.js:70]:
```suggestion
            border='1px dashed'
            borderColor={hovering ? 'gray.900' : 'gray.200'}
            borderRadius='6px'
```

File: src/app/src/components/ModalSections/FileUpload.js [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+                justify='center'
+            >
+                <CloudIconWithBackground />
+                <Text mb={2} fontFamily="'Inter', san-serif">
+                    Drag &amp; drop files here or
+                </Text>
+                <Button
+                    mb={6}
+                    color='gray.600'
+                    bg='gray.200'
```
Comment (rajadain) [inline src/app/src/components/ModalSections/FileUpload.js:87]: Let's use a variant here too, from the wireframes:

```suggestion
                    variant='primary'
```

```diff
diff --git a/src/app/src/theme.js b/src/app/src/theme.js
index 5a5af18..7b33e14 100644
--- a/src/app/src/theme.js
+++ b/src/app/src/theme.js
@@ -18,6 +18,19 @@ const Button = {
                 bg: 'gray.400',
             },
         },
+        primary: {
+            bg: 'gray.100',
+            color: 'gray.600',
+            _hover: {
+                bg: 'gray.600',
+                textDecoration: 'underline',
+                color: 'white',
+            },
+            _disabled: {
+                bg: 'gray.400',
+                color: 'white',
+            },
+        },
         secondary: {
             bg: 'white',
             borderWidth: '1px',
```

**Review by rajadain — COMMENTED** [review]
Some style fixes for the File Upload

File: src/app/src/components/ModalSections/HowItWorks.js (hunk 3 above)
Comment (mstone121) [reply src/app/src/components/ModalSections/HowItWorks.js]: adccd6d04734c4768e440641d559a2bd3fd81bd0

File: src/app/src/components/ModalSections/Introduction.js (hunk 4 above)
Comment (mstone121) [reply src/app/src/components/ModalSections/Introduction.js]: 36534bf6c6c65872deb171434e8d77535ccd2ac3

File: src/app/src/components/ModalSections/ModalSection.js (hunk 5 above)
Comment (mstone121) [reply src/app/src/components/ModalSections/ModalSection.js]: 36534bf6c6c65872deb171434e8d77535ccd2ac3

File: src/app/src/components/ModalSections/Introduction.js (hunk 6 above)
Comment (mstone121) [reply src/app/src/components/ModalSections/Introduction.js:62]: a08299d6da565b611edb06ab7f0cad62fcef476c

File: src/app/src/components/ModalSections/FileUpload.js (hunk 7 above)
Comment (mstone121) [reply src/app/src/components/ModalSections/FileUpload.js:70]: 677c4564229cb809ad9ac48dcbedd3355794741c

File: src/app/src/components/ModalSections/FileUpload.js (hunk 8 above)
Comment (mstone121) [reply src/app/src/components/ModalSections/FileUpload.js]: 5e26aac70909e3b39a6f05a2af87c0487d249eec

File: src/app/src/theme.js [hunk 9]
```diff
@@ -1,21 +1,89 @@
...
 };
 
 const theme = extendTheme({
     components: {
         Button,
+        Heading,
     },
     fonts: {
         heading: `'Quicksand', sans-serif`,
         body: `'Quicksand', sans-serif`,
```
Comment (rajadain) [inline src/app/src/theme.js:72]: Let's switch the body font to Inter, as we're referencing the heading font directly for `welcomeIntro`.

```suggestion
        body: `'Inter', sans-serif`,
```

**Review by rajadain — COMMENTED** [review]
Excellent work! Two final comments before this is ready to merge:

1. Let's change the body font to Inter, as implied in the wireframes
2. Splitting utility functions into their own files feels premature at this time. Let's make a single `utils.js` file and keep the functions there for now. If we start to see patterns in the functions, we can split them into separate files then.

**Review by rajadain — APPROVED** [review]
+1 great work! This looks and works great.

The `react-remove-scroll-bar` issue seems to be related to having multiple modals open: https://github.com/chakra-ui/chakra-ui/issues/6213, would recommend keeping an eye on that issue and implementing any fix that may be posted there in the future.

This is good to merge!

**Pull request merged**